Semantics.md

```diff
@@ -157,6 +162,146 @@ Store operators do not produce a value.
 
 The above operators operate on the [default linear memory](#linear-memory).
 
+### Atomic Memory Accesses
+
+Atomic memory accesses are separated into three categories, load/store,
+read-modify-write, and compare-exchange. All atomic memory accesses require a
+shared linear memory. Attempting to use atomic access operators on non-shared
+linear memory is a validation error.
+
+Currently all atomic memory accesses are [sequentially
+consistent](https://en.wikipedia.org/wiki/Sequential_consistency). This
+restriction may be relaxed in the future.
+
+Atomic load/store memory accesses behave like their non-atomic counterparts,
+with the exception that the ordering of accesses is sequentially consistent.
+
+  * `i32.atomic.load8_s`: atomically load 1 byte and sign-extend i8 to i32
+  * `i32.atomic.load8_u`: atomically load 1 byte and zero-extend i8 to i32
+  * `i32.atomic.load16_s`: atomically load 2 bytes and sign-extend i16 to i32
+  * `i32.atomic.load16_u`: atomically load 2 bytes and zero-extend i16 to i32
+  * `i32.atomic.load`: atomically load 4 bytes as i32
+  * `i64.atomic.load8_s`: atomically load 1 byte and sign-extend i8 to i64
+  * `i64.atomic.load8_u`: atomically load 1 byte and zero-extend i8 to i64
+  * `i64.atomic.load16_s`: atomically load 2 bytes and sign-extend i16 to i64
+  * `i64.atomic.load16_u`: atomically load 2 bytes and zero-extend i16 to i64
+  * `i64.atomic.load32_s`: atomically load 4 bytes and sign-extend i32 to i64
+  * `i64.atomic.load32_u`: atomically load 4 bytes and zero-extend i32 to i64
+  * `i64.atomic.load`: atomically load 8 bytes as i64
+  * `f32.atomic.load`: atomically load 4 bytes as f32
+  * `f64.atomic.load`: atomically load 8 bytes as f64
+  * `i32.atomic.store8`: wrap i32 to i8 and atomically store 1 byte
+  * `i32.atomic.store16`: wrap i32 to i16 and atomically store 2 bytes
+  * `i32.atomic.store`: (no conversion) atomically store 4 bytes
+  * `i64.atomic.store8`: wrap i64 to i8 and atomically store 1 byte
+  * `i64.atomic.store16`: wrap i64 to i16 and atomically store 2 bytes
+  * `i64.atomic.store32`: wrap i64 to i32 and atomically store 4 bytes
+  * `i64.atomic.store`: (no conversion) atomically store 8 bytes
+  * `f32.atomic.store`: (no conversion) atomically store 4 bytes
+  * `f64.atomic.store`: (no conversion) atomically store 8 bytes
+
+Atomic read-modify-write (RMW) operators atomically read a value from a memory
+index, modify the value, and store the resulting value to the same memory
+index. All RMW operators return the value read from memory before the modify
+operation was performed.
+
+The sign-agnostic operations are further described [below](#32-bit-integer-operators).
+
+| Name | Read | Modify | Write |
+| ---- | ---- | ---- | ---- |
+| `i32.atomic.rmw8_s.add` | 1 byte, sign-extended i8 to i32 | sign-agnostic addition | 1 byte, wrapped from i32 to i8 |
```

Comment:
The cmpxchg clarification looks good. So good, that I'm now wondering about a Return column for the rmw table too :-). This is admittedly a really minor nit at this point because none of the current rmw operators care whether their inputs are extended or wrapped, but if we add other rmw operators in the future, they could potentially care.